bob/src/cluster/quorum.rs

```diff
@@ -239,6 +245,64 @@ impl Quorum {
             Ok(())
         }
     }
+
+    pub(crate) async fn delete_aliens(
+        &self,
+        mut failed_nodes: Vec<String>,
+        key: BobKey,
+    ) -> Result<(), Error> {
+        if failed_nodes.is_empty() {
+            warn!(
+                "DELETE[{}] trying to aliens, but there are no failed nodes: unreachable",
+                key
+            );
+            return Err(Error::internal());
+        }
+        trace!("selection of free nodes available for data writing");
+        let (primary, secondary) = self.mapper.get_support_nodes(key, failed_nodes.len());
+        debug!(
+            "DELETE[{}] sup delete nodes, primary: {:?} secondary: {:?}",
+            key, &primary, &secondary
+        );
+        let primary_need_remote_backup: Vec<_> = failed_nodes.drain(..primary.len()).collect();
+        let secondary_need_remote_backup: Vec<_> = failed_nodes.drain(..secondary.len()).collect();
```

Comment:
`secondary` is ALL cluster nodes except `primary` and `local`! Your modification of `get_support_nodes` is broken!
You can get list of nodes from `mapper` right here and filter out `primary` and `local` also right here